Mage.Sets/src/mage/cards/c/CaseOfThePilferedProof.java

```diff
@@ -0,0 +1,175 @@
+
+    CaseOfThePilferedProofTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new AddCountersTargetEffect(CounterType.P1P1.createInstance()).setText("put a +1/+1 counter on it"));
+        this.setTriggerPhrase("Whenever a Detective enters the battlefield under your control and whenever a Detective you control is turned face up, ");
+    }
+
+    private CaseOfThePilferedProofTriggeredAbility(final CaseOfThePilferedProofTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public CaseOfThePilferedProofTriggeredAbility copy() {
+        return new CaseOfThePilferedProofTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        switch (event.getType()) {
+            case ENTERS_THE_BATTLEFIELD:
+            case TURNED_FACE_UP:
+                return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (event.getType() == GameEvent.EventType.ENTERS_THE_BATTLEFIELD ||
+                event.getType() == GameEvent.EventType.TURNED_FACE_UP) {
+            Permanent permanent = game.getPermanent(event.getSourceId());
+            if (permanent != null
+                    && permanent.hasSubtype(SubType.DETECTIVE, game)
+                    && permanent.isControlledBy(this.getControllerId())) {
+                for (Effect effect : getEffects()) {
+                    effect.setTargetPointer(new FixedTarget(event.getTargetId(), game));
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+}
+
+class CaseOfThePilferedProofReplacementEffect extends ReplacementEffectImpl {
+
+    CaseOfThePilferedProofReplacementEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Benefit);
+        this.staticText = "If one or more tokens would be created under your control, those tokens plus a Clue token are created instead";
+    }
+
+    private CaseOfThePilferedProofReplacementEffect(final CaseOfThePilferedProofReplacementEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public CaseOfThePilferedProofReplacementEffect copy() {
+        return new CaseOfThePilferedProofReplacementEffect(this);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.CREATE_TOKEN;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        return source.isControlledBy(event.getPlayerId());
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        if (event instanceof CreateTokenEvent) {
+            CreateTokenEvent tokenEvent = (CreateTokenEvent) event;
+            Map<Token, Integer> tokens = tokenEvent.getTokens();
+            tokens.put(new ClueArtifactToken(), 1);
```

Comment:
just checking, does this still work when the token being created is a clue already? I think they're unique objects so should be fine

Reply:
Yes, this works as expected

Reply:
All other cards with same logic search an existing token and change amount. So it must use same code style.

Search by `Event.getTokens();`:

![shot_240201_065038](https://github.com/magefree/mage/assets/8344157/3c7a858d-50a5-4ef0-b092-30ead19f60cb)